cmd/object-api-common.go

```diff
@@ -313,6 +313,50 @@ func listObjectsNonSlash(ctx context.Context, bucket, prefix, marker, delimiter
 	return result, nil
 }
 
+func fsWalk(ctx context.Context, obj ObjectLayer, bucket, prefix string, doneCh chan struct{}, listDir ListDirFunc, getObjInfo func(context.Context, string, string) (ObjectInfo, error), getObjectInfoDirs ...func(context.Context, string, string) (ObjectInfo, error)) <-chan ObjectInfo {
```

Comment:
Doc + remove `doneCh`.

Reply:
I won't remove doneCh as stated, context is not used with these requirements..and I am not planning to add it as part of this change. We need to comprehensively do it later for certain operations.

Currently Walk() caller is just Lifecycle where there is no premature doneCh. The doneCh is associated with either lifecycle completes a full round. 

This is why context doesn't make sense which is meant for a timeout of a caller. We don't necessarily know when the context shall be closed in case of Walk().